### PR TITLE
changing existing commands, having backward compactibility

### PR DESCRIPTION
**Pull request by ymishrapalo1992**
## Status

- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-16954

## Description

Ships four backward-compatible extensions on existing **GoogleDrive** pack commands so the SaaS remediation playbooks (Soft Delete, Admin/System Quarantine, User Quarantine, Change Sharing) can compose entirely from pack primitives without any net-new public commands.

Every change is opt-in (no `defaultValue` on the new args, no removal or rename of any existing context path). Existing playbooks observe bit-identical request URLs and response shapes when the new arguments are omitted.

### What changed

| ID | Surface | Change | Default behavior |
|:---|:---|:---|:---|
| **E1** | `!google-drive-file-delete` | New `soft_delete` boolean arg. When `true`, calls Drive v3 `files.update` (`PATCH /files/{id}`) with body `{"trashed": true}` instead of the existing permanent `DELETE`. | `false`/omitted → today's hard-delete behavior is preserved. |
| **E2** | `!google-drive-file-permission-create` | New `transfer_ownership` and `move_to_new_owners_root` boolean args. Forwarded as the documented `transferOwnership` and `moveToNewOwnersRoot` query parameters on `permissions.create`. | omitted → query parameters are not appended to the URL. |
| **E3** | `!google-drive-file-permission-delete` | New `ignore_not_found` boolean arg. When `true`, a Not Found response from the vendor is treated as success and surfaced via a new `alreadyRemoved` output flag. | `false`/omitted → today's "raise on Not Found" behavior is preserved. |
| **E4** | `!google-drive-file-permissions-list` | Declared the `permissionDetails.{permissionType,inheritedFrom,role,inherited}` outputs that Drive already returns when `fields=*` is requested (yml-only — no Python change). | n/a — purely additive context paths. |

### Backward-compatibility math

- **0** new public commands
- **0** deprecations
- **0** context-path renames
- **4** additive optional args (none have a `defaultValue` — silent defaults are explicitly avoided per the pack's coding standard)
- **9** additive outputs

### Vendor-side spec (cross-checked against the Google Drive v3 Discovery document)

Every change was verified against `https://www.googleapis.com/discovery/v1/apis/drive/v3/rest` (the machine-readable source of truth for the Drive API):

- `files.update` (PATCH) accepts a body with the writable `trashed` boolean. Note: `trashedTime` is documented as **only populated for items in shared drives** — surfaced in the yml output description so playbook authors don't write null-unsafe filters.
- `permissions.create` accepts the GA `transferOwnership` and `moveToNewOwnersRoot` query parameters; without `transferOwnership=true`, posting `role=owner` on an existing file is rejected by Drive with `400 ownershipChangeAcrossDomainNotPermitted` even for in-domain transfers.
- `permissions.delete` returns `204 No Content` on success and `404 notFound` when the permission has already been removed.
- `Permission.permissionDetails[].inherited` is documented as always populated when the request asks for `fields=*` — which the integration already does.

### Implementation notes

- **v3 over v2 for E1.** The trash action could have used `PATCH /drive/v2/files/{id}/trash`, but the rest of the pack is v3-only — introducing a v2 caller would add a new URL_SUFFIX entry and a partial second API version for zero functional benefit. The v3 `{"trashed": true}` PATCH has the same audit footprint (the Drive activity log records `trash`, not `delete`).
- **`move_to_new_owners_root` shipped alongside `transfer_ownership` (E2).** Same call site, no extra plumbing — saves a follow-up MR for the documented companion query parameter.
- **E3 detects `404` by string-matching `"Status: 404"`** in the `DemistoException` message produced by the shared `GSuiteApiModule.COMMON_MESSAGES["HTTP_ERROR"]` template. The alternative (plumbing a `status_code` attribute through `GSuiteApiModule`) was deferred because the blast radius spans every G-Suite integration. A pinned unit test injects the literal `"Status: 404. Reason: Not Found"` substring — if the shared module's template ever changes, the test fails loudly.

### Why no new public commands

Every primitive required by the four downstream remediation flows is already exposed by existing pack commands (`!google-drive-file-get`, `!google-drive-file-move`, `!google-drive-file-create`, `!google-drive-file-permissions-list`, `!google-drive-file-permission-create`, `!google-drive-file-permission-delete`). Composing these in playbooks — gated on the four backward-compatible extensions in this MR — is more flexible than shipping opinionated orchestrator commands, avoids duplicating context paths
### Files changed

| Path | Why |
|:---|:---|
| `Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml` | E1/E2/E3 args + E1/E3/E4 outputs (+56 lines). |
| `Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py` | E1/E2/E3 Python wiring across three command handlers (+62 lines); Google-style docstrings. |
| `Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py` | 7 new unit tests covering success, default-preserved, and the 404/non-404 branches of E3 (+276 lines). All tests mock `GSuiteApiModule.GSuiteClient.http_request`; no live API calls. |
| `Packs/GoogleDrive/Integrations/GoogleDrive/README.md` | Regenerated via `demisto-sdk generate-docs`. |
| `Packs/GoogleDrive/Integrations/GoogleDrive/command_examples.txt` | +4 invocations covering each extension. |
| `Packs/GoogleDrive/ReleaseNotes/1_5_0.md` | 4 customer-facing bullets. |
| `Packs/GoogleDrive/pack_metadata.json` | `currentVersion`: `1.4.0` → `1.5.0`. |

### Verification

| Command | Result |
|:---|:---|
| `demisto-sdk format -i Packs/GoogleDrive --no-graph -y` | ✅ |
| `demisto-sdk validate -i Packs/GoogleDrive` | ✅ All validations passed. |
| `demisto-sdk pre-commit -i Packs/GoogleDrive` | ✅ Every hook green: pylint, mypy, **pytest** (7 new + all existing), ruff-format, ruff-py3.10, ruff-py3.12, xsoar-lint, markdownlint, secrets, MDX, brack. |
| `demisto-sdk generate-docs -i Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml -e ...command_examples.txt --no-graph` | ✅ README sections replaced for all 4 touched commands. |

## Must have

- [x] Tests — 7 new unit tests cover happy path, default-preserved (no behavioral drift), and edge cases (E3's 404 swallow vs. non-404 re-raise).
- [x] Documentation — README regenerated by `demisto-sdk generate-docs`; arg/output descriptions explain how to obtain ID values and the vendor-side caveats (`trashedTime` only on shared drives, `move_to_new_owners_root` only effective when `transfer_ownership=true`).
- [x] Version bump — pack `currentVersion` bumped minor (1.4.0 → 1.5.0).

relates: link to the issue

### CODE REVIEW

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 1]
```diff
@@ -2604,10 +2604,22 @@ script:
       predefined:
       - "true"
       - "false"
+    - auto: PREDEFINED
+      description: 'Whether to move the file to Trash instead of permanently deleting it. When set, the file is patched with trashed=true rather than issuing a hard delete. Omitting this argument preserves the existing hard-delete behavior.'
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: 'Whether to move the file to Trash instead of permanently deleting it. When set to "true", the file is patched with trashed=true rather than issuing a hard delete. Omitting this argument preserves the existing hard-delete behavior.'
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 2]
```diff
@@ -2604,10 +2604,22 @@ script:
...
+      name: soft_delete
+      predefined:
+      - "true"
+      - "false"
     outputs:
     - contextPath: GoogleDrive.File.File.id
       description: ID of the deleted file.
       type: String
+    - contextPath: GoogleDrive.File.File.trashed
+      description: Whether the file is in Trash after a soft delete. Only populated when soft_delete=true.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: Whether the file is in Trash after a soft delete. Only populated when soft_delete is set to "true".
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 3]
```diff
@@ -2761,7 +2797,7 @@ script:
       description: A link to the user's profile photo, if available.
       type: String
   - name: google-drive-file-permission-delete
-    description: Delete a permission.
+    description: Delete a permission. When the ignore_not_found argument is set to true, a Not Found response from the vendor is treated as success and surfaced via the alreadyRemoved output, which is useful inside loops over a previously-listed permission set.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
    description: Deletes a permission. When the ignore_not_found argument is set to "true", a Not Found response from the vendor is treated as success and surfaced via the alreadyRemoved output, which is useful inside loops over a previously-listed permission set.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 4]
```diff
@@ -2604,10 +2604,22 @@ script:
...
       - "false"
+    - auto: PREDEFINED
+      description: 'Whether to move the file to Trash instead of permanently deleting it. When set, the file is patched with trashed=true rather than issuing a hard delete. Omitting this argument preserves the existing hard-delete behavior.'
+      name: soft_delete
+      predefined:
+      - "true"
+      - "false"
     outputs:
     - contextPath: GoogleDrive.File.File.id
       description: ID of the deleted file.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: The ID of the deleted file.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1886]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2341]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2382]:
```suggestion
      description: The ID of the file to replace. Can be retrieved using the `google-drive-files-list` command.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 5]
```diff
@@ -2591,7 +2591,7 @@ script:
       description: A link to the user's profile photo, if available.
       type: String
   - name: google-drive-file-delete
-    description: Permanently deletes a file owned by the user without moving it to the trash. If the file belongs to a shared drive the user must be an organizer on the parent. If the target is a folder, all descendants owned by the user are also deleted.
+    description: Permanently deletes a file owned by the user without moving it to the trash. If the file belongs to a shared drive the user must be an organizer on the parent. If the target is a folder, all descendants owned by the user are also deleted. When the soft_delete argument is set to true, the file is moved to Trash instead of being permanently deleted.
     arguments:
     - name: file_id
       description: ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2688]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2762]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 6]
```diff
@@ -2761,7 +2797,7 @@ script:
       description: A link to the user's profile photo, if available.
       type: String
   - name: google-drive-file-permission-delete
-    description: Delete a permission.
+    description: Delete a permission. When the ignore_not_found argument is set to true, a Not Found response from the vendor is treated as success and surfaced via the alreadyRemoved output, which is useful inside loops over a previously-listed permission set.
     arguments:
     - name: file_id
       description: ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2839]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3085]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3150]:
```suggestion
      description: The ID of the requested file. Can be retrieved using the `google-drive-files-list` command.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1614]:
```suggestion
      description: The page token for shared drives.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1836]:
```suggestion
      description: The page token for shared drives.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: The page token for shared drives.
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1620]:
```suggestion
      description: "Whether to issue the request as a domain administrator. If set to "true", all shared drives of the domain in which the requester is an administrator are returned."
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1714]:
```suggestion
      description: "Whether to issue the request as a domain administrator. If set to "true", all shared drives of the domain in which the requester is an administrator are returned."
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1813]:
```suggestion
      description: "Whether to grant the requester access if they are an administrator of the domain to which the shared drive belongs."
```

Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: "Whether to issue the request as a domain administrator. If set to "true", all shared drives of the domain in which the requester is an administrator are returned."
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Google Drive
+
+- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new ***soft_delete*** argument, in addition to the existing permanent-delete behavior.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument, in addition to the existing permanent delete behavior.
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new ***transfer_ownership*** and ***move_to_new_owners_root*** arguments. Use these when promoting a user to file owner.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new *transfer_ownership* and *move_to_new_owners_root* arguments. Use these when promoting a user to file owner.
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+- Updated the ***google-drive-file-permission-delete*** command with the new ***ignore_not_found*** argument so that already-removed permissions are treated as success. Useful when removing many permissions in a loop.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-permission-delete*** command with the new *ignore_not_found* argument so that already removed permissions are treated as success. Useful when removing many permissions in a loop.
```

**Review by julieschwartz18 — COMMENTED** [review]
@ymishrapalo1992 Doc review complete, nice job. Please regenerate the integration README file from the yaml file. In the meantime, adding the label docs-approved

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 10]
```diff
@@ -2604,10 +2604,22 @@ script:
       predefined:
       - "true"
       - "false"
+    - auto: PREDEFINED
```
Comment (YuvHayun) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2607]: Let's add default value false.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 11]
```diff
@@ -2694,6 +2718,18 @@ script:
...
+      name: transfer_ownership
+      predefined:
+      - "true"
+      - "false"
+    - auto: PREDEFINED
+      description: 'Whether to move the file to the new owner''s root folder. Only takes effect when transfer_ownership=true.'
+      name: move_to_new_owners_root
+      predefined:
+      - "true"
+      - "false"
```
Comment (YuvHayun) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2735]: add default values.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 12]
```diff
@@ -2776,6 +2812,22 @@ script:
       predefined:
       - "true"
       - "false"
+    - auto: PREDEFINED
+      description: 'Whether to treat a Not Found response from the vendor (for example, when the permission was already removed) as success and surface it via the alreadyRemoved output, instead of raising an error. Useful inside loops over a previously-listed permission set.'
+      name: ignore_not_found
+      predefined:
+      - "true"
+      - "false"
```
Comment (YuvHayun) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:2824]: Add default values.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 13]
```diff
@@ -1393,19 +1393,35 @@ def get_labels_command(client: "GSuiteClient", args: dict[str, str]) -> CommandR
...
+    soft_delete = argToBoolean(args.get("soft_delete") or "false")
+
+    if soft_delete:
+        response = client.http_request(
+            url_suffix=url_suffix,
+            method="PATCH",
+            params=http_request_params,
+            body={"trashed": True},
+        )
+        return handle_response_file_single(response, args)
```
Comment (YuvHayun) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:1423]: I See this require different method and such..
I'd consider just adding a new command rather than updating existing one.
Something like "<prefix>-move-file-to-trash"

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 14]
```diff
@@ -1679,11 +1702,38 @@ def file_permission_delete_command(client: "GSuiteClient", args: dict[str, str])
...
+    except DemistoException as exc:
+        # Only swallow Not Found responses that refer to the *permission* itself
+        # (i.e. the permission was already removed). Not Found responses that
+        # refer to the parent file are operator errors and must still be raised
+        # so playbooks do not silently mask a typo in file_id.
+        # The shared GSuiteApiModule surfaces these as:
+        #   "Not found. Reason: Permission not found: <id>"            -- swallowed
+        #   "Not found. Reason: File not found: <id>"                  -- re-raised
+        #   "HTTP Connection error occurred. Status: 404. Reason: ..." -- re-raised unless body
+        #                                                                 names a permission
```
Comment (YuvHayun) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: remove/shorten

**Review by YuvHayun — COMMENTED** [review]
Hey @ymishrapalo1992 see my comments.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py (hunk 13 above)
Comment (ymishrapalo1992) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:1423]: Thanks for the careful read. I'd like to keep soft_delete as a flag on the existing command rather than splitting into !google-drive-file-trash, for three reasons:

Backward-compat ergonomics for playbooks. Existing playbooks that call !google-drive-file-delete can opt into soft-delete by flipping one argument; with a new command, every retrofit becomes a node replacement.
AGENTS.md §0 explicitly says: "NEVER duplicate an existing Demisto command — extend or reuse it. Adding a second command for the same action breaks playbooks and context paths." I read "the same action" broadly here — both verbs operate on the same logical action (remove a file from the user's view).
No precedent in sibling packs. MicrosoftGraphFiles ships only !msgraph-delete-file; there's no !msgraph-trash-file for me to mirror.
The flag default is false, the new branch is fully covered by unit tests, and the existing hard-delete path is untouched.

Comment (YuvHayun) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:1423]: Okay, it was a suggestion.
Please let me know when the rest of the fixes are done and ensure the pre-commit is passing.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 15]
```diff
@@ -2604,22 +2604,35 @@ script:
       predefined:
       - "true"
       - "false"
+    - auto: PREDEFINED
+      defaultValue: "false"
+      description: 'Whether to move the file to Trash instead of permanently deleting it. When set to "true", the file is patched with trashed=true rather than issuing a hard delete. Omitting this argument preserves the existing hard-delete behavior.'
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]:
```suggestion
      description: 'Whether to move the file to the Trash instead of permanently deleting it. If "true", the file is moved to the Trash. If "false", the file is permanently deleted.'
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 16]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Google Drive
+
+- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument, in addition to the existing permanent delete behavior.
+- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new *transfer_ownership* and *move_to_new_owners_root* arguments. Use these when promoting a user to file owner.
+- Updated the ***google-drive-file-permission-delete*** command with the new *ignore_not_found* argument so that already removed permissions are treated as success. Useful when removing many permissions in a loop.
+- Updated the ***google-drive-file-permissions-list*** command to surface whether each permission is inherited and where it is inherited from.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-permissions-list*** command to indicate whether each permission is inherited and from where it is inherited.
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 17]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Google Drive
+
+- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument, in addition to the existing permanent delete behavior.
+- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new *transfer_ownership* and *move_to_new_owners_root* arguments. Use these when promoting a user to file owner.
+- Updated the ***google-drive-file-permission-delete*** command with the new *ignore_not_found* argument so that already removed permissions are treated as success. Useful when removing many permissions in a loop.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-permission-delete*** command with the new *ignore_not_found* argument, which allows already removed permissions to be treated as a success.
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Google Drive
+
+- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument, in addition to the existing permanent delete behavior.
+- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new *transfer_ownership* and *move_to_new_owners_root* arguments. Use these when promoting a user to file owner.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-permission-create*** command to support ownership transfer via the new *transfer_ownership* and *move_to_new_owners_root* arguments when promoting a user to file owner.
```

File: Packs/GoogleDrive/ReleaseNotes/1_5_0.md [hunk 19]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Google Drive
+
+- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument, in addition to the existing permanent delete behavior.
```
Comment (julieschwartz18) [inline Packs/GoogleDrive/ReleaseNotes/1_5_0.md]:
```suggestion
- Updated the ***google-drive-file-delete*** command to support moving a file to Trash via the new *soft_delete* argument.
```

**Review by julieschwartz18 — COMMENTED** [review]
@ymishrapalo1992 Doc review complete, please check the comments and regenerate the integration README file. docs-approved label already applied

**Comment by ymishrapalo1992** [comment]
@julieschwartz18 comments have been addressed , can you please re review ?

**Review by YuvHayun — APPROVED** [review]
